src/main.rs

```diff
@@ -49,10 +49,23 @@ struct CalypsoArgs {
 fn read_can(pub_path: &str, can_interface: &str) -> JoinHandle<u32> {
     //open can socket channel at name can_interface
     let mut client = MqttClient::new(pub_path, "calypso-decoder");
-    client.connect().expect("Could not connect to Siren!");
+    if client.connect().is_err() {
+        println!("Unable to connect to Siren, going into reconnection mode.");
+        if client.reconnect().is_ok() {
+            println!("Reconnected to Siren!");
+        }
+    }
+    
```

Comment:
are you sure this block of code is needed?  What if we just ignore an error, and then reconnect below in line 62 as that is looping?